packages/@o3r/design/src/core/design-token/renderers/design-token-style.renderer.ts

```diff
@@ -80,6 +98,31 @@ export const getTokenSorterByRef: TokenListTransform = (variableSet) => {
   };
 };
 
+/**
+ * Retrieve default file writer (based on Node `fs.promise.writeFile` interface)
+ * @param existsFile Function determining if the file exists
+ * @param logger Custom Logger
+ * @returns
+ */
+export const getDefaultFileWriter = (existsFile: (file: string) => boolean, logger?: Logger): typeof fs.writeFile => {
+  return async (file, ...args) => {
+    // eslint-disable-next-line @typescript-eslint/no-base-to-string -- The file path is always proposing the `toString` method
+    const fileString = file.toString();
+    const { writeFile, mkdir } = await import('node:fs/promises');
+    if (!existsFile(fileString)) {
+      const { dirname } = await import('node:path');
+      try {
+        await mkdir(dirname(fileString), { recursive: true });
+      } catch {
+        // ignore folder creation failure
+      }
+    }
```

Comment:
The file may not exist but the folder can 
why not doing 
```suggestion
      const { dirname } = await import('node:path');
      try {
        await mkdir(dirname(fileString), { recursive: true });
      } catch {
        // ignore folder creation failure
      }
```